File: 프로그래머스/2/135807. 숫자 카드 나누기/숫자 카드 나누기.js
function solution(arrayA, arrayB) {
  const gcd = (a, b) => (a % b === 0 ? b : gcd(b, a % b))

  let resultA = arrayA[0]
  let resultB = arrayB[0]

  arrayA.forEach((el) => resultA = gcd(resultA, el))
  arrayB.forEach((el) => resultB = gcd(resultB, el))

  let chkA = true
  let chkB = true

  arrayA.forEach((el) => {if (el % resultB === 0) chkA = false})
  arrayB.forEach((el) => {if (el % resultA === 0) chkB = false})

  return chkA && chkB ? (resultA > resultB ? resultA : resultB) : chkA ? resultB : chkB ? resultA : 0
}

File: 프로그래머스/2/135807. 숫자 카드 나누기/숫자 카드 나누기.py
def GCD(x, y):
    while y: 
        x,y = y, x % y
    return x
 
def notDiv(arr, gcd):
    for n in arr:
        if n % gcd == 0: return False
    return True

def solution(arrayA, arrayB):
    answer = 0
    gcdA = arrayA[0]
    gcdB = arrayB[0]
    
    for a in arrayA[1:]: gcdA = GCD(a, gcdA)
    for b in arrayB[1:]:gcdB = GCD(b, gcdB)
        
    if notDiv(arrayA, gcdB): answer = max(answer, gcdB)
    if notDiv(arrayB, gcdA):answer = max(answer, gcdA)
        
    return answer